include/pagmo/utils/generic.hpp

```diff
@@ -265,7 +266,7 @@ std::vector<std::vector<vector_double::size_type>> kNN(const std::vector<vector_
         std::vector<vector_double::size_type> idxs(N);
         std::iota(idxs.begin(), idxs.end(), vector_double::size_type(0u));
         std::sort(idxs.begin(), idxs.end(), [&distances](vector_double::size_type idx1, vector_double::size_type idx2) {
-            return distances[idx1] < distances[idx2];
+            return detail::less_than_f(distances[idx1], distances[idx2]);
```

Comment:
They are indexes, but they are ordered w.r.t. the associated  distance which is a float